js/ui/date_box/ui.date_box.strategy.calendar.js

```diff
@@ -107,47 +113,57 @@ const CalendarStrategy = DateBoxStrategy.inherit({
         this.dateBox.setAria('activedescendant', e.actionValue);
     },
 
-    popupConfig: function(popupConfig) {
-        const toolbarItems = popupConfig.toolbarItems;
+    _getTodayButtonConfig() {
         const buttonsLocation = this.dateBox.option('buttonsLocation');
+        const isButtonsLocationDefault = buttonsLocation === 'default';
 
-        let position = [];
+        const position = isButtonsLocationDefault ? ['bottom', 'center'] : splitPair(buttonsLocation);
 
-        if(buttonsLocation !== 'default') {
-            position = splitPair(buttonsLocation);
-        } else {
-            position = ['bottom', 'center'];
-        }
+        return {
+            widget: 'dxButton',
+            toolbar: position[0],
+            location: position[1] === 'after' ? 'before' : position[1],
+            options: {
+                onInitialized: function(e) {
+                    e.component.registerKeyHandler('escape', this._escapeHandler.bind(this));
+                }.bind(this),
+                onClick: (args) => { this._widget._toTodayView(args); },
+                text: this.dateBox.option('todayButtonText'),
+                type: 'today',
+            }
+        };
+    },
 
-        if(this.dateBox.option('applyValueMode') === 'useButtons' && this._isCalendarVisible()) {
-            toolbarItems.unshift({
-                widget: 'dxButton',
-                toolbar: position[0],
-                location: position[1] === 'after' ? 'before' : position[1],
-                options: {
-                    onInitialized: function(e) {
-                        e.component.registerKeyHandler('escape', this._escapeHandler.bind(this));
-                    }.bind(this),
-                    onClick: (args) => { this._widget._toTodayView(args); },
-                    text: messageLocalization.format('dxCalendar-todayButtonText'),
-                    type: 'today'
-                }
-            });
+    _isCalendarVisible: function() {
+        const isOptionsEmpty = isEmptyObject(this.dateBox.option('calendarOptions'));
+        const isVisible = () => this.dateBox.option('calendarOptions.visible') !== false;
+
+        return isOptionsEmpty || isVisible();
+    },
+
+    _getPopupToolbarItems(toolbarItems) {
+        const applyModeUsedButtons = this.dateBox.option('applyValueMode') === 'useButtons';
+        const usedButtonsAndIsVisible = applyModeUsedButtons && this._isCalendarVisible();
```

Comment:
We see that it is a "usedButtonsAndIsVisible" when we read `useButtons && this._isCalendarVisible()`. Additional variable with such name only complicated the code. 

If u wanna separate condition to a variable to make code cleaner u can name this like this:
```suggestion
        const shouldRenderTodayButton = useButtons && this._isCalendarVisible();
```

Reply:
https://github.com/DevExpress/DevExtreme/pull/22991/commits/6db8ca42fc7d248ad8eb6711a8299e46aae5347b